st2actions/tests/unit/test_paramiko_ssh.py

```diff
@@ -802,3 +802,33 @@ def test_use_ssh_config_port_value_provided_in_the_config(self, mock_sshclient):
 
         call_kwargs = mock_client.connect.call_args[1]
         self.assertEqual(call_kwargs['port'], 9999)
+
+    @patch.object(ParamikoSSHClient, '_is_key_file_needs_passphrase',
+                  MagicMock(return_value=False))
+    def test_socket_closed(self):
+        conn_params = {'hostname': 'dummy.host.org',
+                       'username': 'ubuntu',
+                       'password': 'pass',
+                       'timeout': '600'}
+        ssh_client = ParamikoSSHClient(**conn_params)
+        ssh_client.connect()
```

Comment:
```suggestion
```

Oh, don't call this - the tiny change you made in the `_connect` function is insignificant to its behavior and should be reverse compatible, and therefore covered by all of the other tests for `ParamikoSSHClient`, so you don't need to test it explicitly.

This test should simply initialize the `ParamikoSSHClient` with some dummy parameters, replace its internals with mocked objects, call `.close()` and run your assertions, as you have done.